[custom_op] Create a new torch._custom_op namespace

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #101824
* __->__ #101823
* #101822

torch/custom_op.py is getting long, and the autograd pieces are going to
make it even longer. I'm planning on just organizing the files under
a torch/_custom_op folder.

Note that the imports now look a bit crazy (from torch._custom_op.impl
import...) but they will look more OK when we figure out the plan to
make custom_op public (coming later).

zou3519 committed:
[custom_op] Create a new torch._custom_op namespace

torch/custom_op.py is getting long, and the autograd pieces are going to
make it even longer. I'm planning on just organizing the files under
a torch/_custom_op folder.

Note that the imports now look a bit crazy (from torch._custom_op.impl
import...) but they will look more OK when we figure out the plan to
make custom_op public (coming later).

[ghstack-poisoned]

albanD reviewed (approved):
Sounds like an ok temp solution!

zou3519 committed:
Update on "[custom_op] Create a new torch._custom_op namespace"


torch/custom_op.py is getting long, and the autograd pieces are going to
make it even longer. I'm planning on just organizing the files under
a torch/_custom_op folder.

Note that the imports now look a bit crazy (from torch._custom_op.impl
import...) but they will look more OK when we figure out the plan to
make custom_op public (coming later).

[ghstack-poisoned]

zou3519 committed:
Update on "[custom_op] Create a new torch._custom_op namespace"


torch/custom_op.py is getting long, and the autograd pieces are going to
make it even longer. I'm planning on just organizing the files under
a torch/_custom_op folder.

Note that the imports now look a bit crazy (from torch._custom_op.impl
import...) but they will look more OK when we figure out the plan to
make custom_op public (coming later).

[ghstack-poisoned]
